### ISSUE TITLE
clipboard function has no error exception

### ISSUE THREAD

**Issue by yms2772** [fyne-io/fyne#650]
### Describe the bug:

When I capture a screenshot, it says
`GLFW: An invalid error was not accepted by the caller: FormatUnavailable: Win32: Failed to convert clipboard to string`
`GLFW: Please report this bug in the Go package immediately.`
and exit program immediately.

### Example code:

```go
    
    func keyCheckRealTime(clp string) (bool, string) {
	isMatched, _ := regexp.MatchString(`\w{8}-\w{4}-\w{4}-\w{4}-\w{12}`, clp)

	if isMatched {
		return true, clp
	}

	return false, clp
    }

    clpStatus, clp := keyCheckRealTime(w.Clipboard().Content())
    if clpStatus {
        ok := dialog.NewConfirm("test","Do you want to use it? ", func(res bool) {
        if res {
                keyEntry.SetText(clp)
        }
    }, w)
        ok.SetConfirmText("Yes")
        ok.SetDismissText("No")
        ok.Show()
        
        break
        }
```

### Device (please complete the following information):
 - OS: Windows
 - Version: Windows 10
 - Go version: 1.13
 - Fyne version: 1.2.2

**Comment by andydotxyz** [comment]
Unfortunately the toolkit only supports unicode strings in copy and paste at this time.
The constraint is imposed by the GLFW library that we use. To overcome this we would need to improve it upstream or write our own clipboard integration.

**Comment by zecman** [comment]
+1 for this.

right now i cannot use it because GLFW panics and i have no chance to handle this with fyne.

**Comment by changkun** [comment]
The current Clipboard interface assumes data is always a string:

https://github.com/fyne-io/fyne/blob/c29a0624ed96ba1b8f45d903b6941824d50e0502/clipboard.go#L4-L9

To address this issue, and support non-UTF8 string, it seems the interface must be changed. What's the plan for this?

**Comment by andydotxyz** [comment]
We have not got a plan yet unfortunately - it depends on the main dependency supporting something other than text.
https://github.com/glfw/glfw/search?q=clipboard+data&type=issues

**Comment by changkun** [comment]
@andydotxyz Yes, but even if glfw supports image data or we implement our own clipboard integration, the existing Clipboard API needs a redesign, which seems falls into non trivial change and can effect many places, such as mobile. Any thoughts on that?

**Comment by andydotxyz** [comment]
It is a significant and breaking change you are right.
We try not to design abstractions until we have some concrete implementations which is why this has not happened yet.
We can design a better API, but until we know what is / will be possible it might be wasted effort.

**Comment by changkun** [comment]
In general, producing a cross-platform clipboard design is very subtle and always involves platform-specific details.
On Linux, X11 clipboard implementation communicates data directly between applications (no global clipboard); On macOS and Windows, there is a global system-wise clipboard, but Windows uses bitmap representation and may have internal color conversion for images, whereas macOS doesn't and only treat as bytes.

The most basic and common for communicating texts is therefore sure, but how much (ability) would need about interacting with a clipboard for fyne users? The answer to this question shapes the eventual design as well as implementation. Specifically, this issue is taking about non-utf8 string, questions are:

1. How many predefined formats (UTF8 texts, images, maybe files, maybe more?) we should define for Clipboard interface? 
2. What are the image formats that should be implemented if we provide methods like `Clipboard.SetImage(image.Image)`?
3. Should the read API be `Clipboard.GetImage() []byte` or `Clipboard.GetImage() image.Image`? The reason we differentiate this is because parsing image encoded bytes is format specific, and also depends on the validity of the content set to clipboard in a thrid party application.
4. Do users need the ability to register a customized clipboard data beyound default offers?

I guess the reason that glfw remains not providing clipboard APIs mosty because the major target users of glfw is GL applications which clipboard is not often needed. This is different from fyne that targeting for building GUI applications, which eventually might needs more sophistication than just rendering a window. Thoughts?

**Comment by andydotxyz** [comment]
Those thoughts are all important, yes. But I don't think we should add these APIs until the drivers can implement them.
